src/leaflet.drawDistortableImage.js

```diff
@@ -100,27 +100,35 @@ L.Draw.DistortableImage = L.Draw.Rectangle.extend({
 
         this._initialLabelText = L.drawLocal.draw.handlers.distortableimage.tooltip.start;
 
+        if (options.createdEventHandler || this.options.createdEventHandler)
+            this._createdEventHandler = options.createdEventHandler ? options.createdEventHandler : this.options.createdEventHandler;
+
         L.Draw.SimpleShape.prototype.initialize.call(this, map, options);
     },
 
     _fireCreatedEvent: function () {
         var that = this,
-            rectangle = new L.Rectangle(this._shape.getBounds(), this.options.shapeOptions).addTo(this._map),
-            popupContent = '<div><label for="imgUrlInput">Enter URL for image</label><input type="text" id="imgUrlInput" name="imgUrlInput"><input type="submit" id="imgUrlSubmit"></div>';
+            rectangle = new L.Rectangle(this._shape.getBounds(), this.options.shapeOptions).addTo(this._map);
 
         this._tempRect = rectangle;
 
-        rectangle.bindPopup(popupContent).openPopup();
-
-        document.getElementById('imgUrlSubmit').onclick = function() {
-            that._onSubmit();
-        };
+        if (this._createdEventHandler)
+        {
+            this._createdEventHandler.call(this);
+        } else {
+            var popupContent = '<div><label for="imgUrlInput">Enter URL for image</label><input type="text" id="imgUrlInput" name="imgUrlInput"><input type="submit" id="imgUrlSubmit"></div>';
+            rectangle.bindPopup(popupContent).openPopup();
+
+            document.getElementById('imgUrlSubmit').onclick = function() {
+                that._onSubmit(document.getElementById('imgUrlInput').value);
+            };
+        }
 
         this._map
             .on('click', this._cancelCreate, this);
     },
 
-    _onSubmit: function() {
+    _onSubmit: function(url) {
```

Comment:
FWIW, I like moving any explicit reference to a DOM element into the same area of the code as that element was created, so if that default DOM interface is modified, all refs to it are contained in that section of the code. It's one step closer to a separation of view and logic.